Bump Android SDK to 3.0.6, match kotlin and gradle plugin versions to…

SurvicatePawelGodlewski opened the pull request:
… match those from SDK, bump targetSdk and compileSdk versions to 34, added Proguard rule for obfuscated xml parsers

dariusz-survicate commented on android/build.gradle:
```diff
@@ -26,7 +26,9 @@ apply plugin: 'com.android.library'
 
 android {
-    compileSdkVersion 31
+    namespace 'com.survicate.flutter'
```
We can remove `package="com.survicate.flutter"` from AndroidManifest

dariusz-survicate replied:
or leave it in manifest and here wrap with `if (project.android.hasProperty("namespace"))`

SurvicatePawelGodlewski replied:
I removed package from Manifest

SurvicatePawelGodlewski commented:
Final changes: 
- bump AGP version to 7.4.2 
- bump Kotlin to 1.8.10
- bump Android Survicate SDK to 3.0.6 (with exclusion of coil 2.5.0)
- added namespace property in build.gradle 
- removed package from manifest
- added proguard rules for xml parsers
